### PR TITLE
Move confusion matrix from %%ml to library.

### PR DESCRIPTION
**Pull request by qimingj**
This is part of efforts to move %%ml magic stuff to library to provide a consistent experience (python only).

### CODE REVIEW

File: datalab/mlalpha/_confusion_matrix.py [hunk 1]
```diff
@@ -11,57 +11,99 @@
...
+  @staticmethod
+  def from_csv(input_csv, headers=None, schema_file=None):
+    """Create a ConfusionMatrix from a csv file.
     Args:
-      predicted_labels: a list of predicted labels.
-      true_labels: a list of true labels.
-      counts: a list of count for each (predicted, true) combination.
+      input_csv: Path to a Csv file (with no header). Can be local or GCS path.
+      headers: Csv headers. If present, it must include 'target' and 'predicted'.
+      schema_file: Path to a JSON file containing BigQuery schema. Used if "headers" is None.
```
Comment (brandondutra) [inline datalab/mlalpha/_confusion_matrix.py]: Add "If present, it must include 'target' and 'predicted'." to the schema_file line too or just say there has to be two columns named target and predicted before the args list.

File: datalab/mlalpha/_confusion_matrix.py [hunk 2]
```diff
@@ -11,57 +11,99 @@
...
-      data.append(predicts_for_current_true_label)
-    self._data = data
+
+    query, _ = data.SqlModule.get_sql_statement_with_environment(sql, {})
+    sql = ('SELECT target, predicted, count(*) as count FROM (%s) group by target, predicted'
+        % query.sql)
+    df = bq.Query(sql).results().to_dataframe()
+    labels = sorted(set(df['target']) | set(df['predicted']))
+    labels_count = len(labels)
+    df['target'] = [labels.index(x) for x in df['target']]
```
Comment (brandondutra) [inline datalab/mlalpha/_confusion_matrix.py]: why don't we just call confusion_matrix() like above?

Comment (qimingj) [inline datalab/mlalpha/_confusion_matrix.py]: We cannot. Scikit-learn's confusion-matrix constructor only accepts a list of "target" and a list of "predicted" (so it doesn't work with large data). It does not accept aggregated data.

File: datalab/mlalpha/_confusion_matrix.py (hunk 1 above)
Comment (qimingj) [inline datalab/mlalpha/_confusion_matrix.py]: Done

**Pull request merged**